File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/server/MatchScoreUpdatePacket.java
package pe.nanamochi.banchus.packets.server;

import java.io.IOException;
import java.io.OutputStream;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.components.ScoreFrame;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.core.ServerPacket;
import pe.nanamochi.banchus.io.IDataWriter;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class MatchScoreUpdatePacket implements ServerPacket {
  private ScoreFrame frame;

  @Override
  public Packets getPacketType() {
    return Packets.BANCHO_MATCH_SCORE_UPDATE;
  }

  @Override
  public void write(IDataWriter writer, OutputStream stream) throws IOException {
    frame.write(writer, stream);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/server/MatchScoreUpdatePacket.kt
package pe.nanamochi.banchus.packets.server

import java.io.OutputStream
import pe.nanamochi.banchus.components.ScoreFrame
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.core.ServerPacket
import pe.nanamochi.banchus.io.DataWriter

class MatchScoreUpdatePacket(var frame: ScoreFrame? = null) : ServerPacket() {
    override val type = PacketType.BANCHO_MATCH_SCORE_UPDATE

    override fun write(writer: DataWriter, out: OutputStream) {
        frame?.write(writer, out)
    }
}
